src/librustc_trans/save/dump_csv.rs

```diff
@@ -1177,10 +1238,15 @@ impl<'l, 'tcx, 'v> Visitor<'v> for DumpCsvVisitor<'l, 'tcx> {
     }
 
     fn visit_stmt(&mut self, s: &ast::Stmt) {
+        let id = s.node.id();
+        if let Some(id) = id {
```

Comment:
I couldn't tell you off the top of my head, but there was definitely a case causing it to panic before I added the None-check.